data/scripts.js

```diff
@@ -2969,8 +2969,31 @@ exports.BattleScripts = {
 
 		// We choose level based on BST. Min level is 70, max level is 99. 600+ BST is 70, less than 300 is 99. Calculate with those values.
 		// Every 10.35 BST adds a level from 70 up to 99. Results are floored. Uses the Mega's stats if holding a Mega Stone
-		// To-do: adjust levels of mons with boosting items (Light Ball, Thick Club etc)
 		var bst = template.baseStats.hp + template.baseStats.atk + template.baseStats.def + template.baseStats.spa + template.baseStats.spd + template.baseStats.spe;
+		// Adjust levels of mons based on abilities (Pure Power, Sheer Force, etc.) and also Eviolite
+		// If a mon's offense stat is increased by M times that stat, with 255 EV, its effective base stat becomes (1+M) * ([base stat] + 50).
+		// If a mon's defensive stat is increased by M times that stat, with 0 EV, its effective base stat becomes (1+M) * ([base stat] + 18).
+		// Cut some ability cases a little slack since many of these mons can't have any other kind of boosting item (based on holistic judgment)
+		var templateAbility = (baseTemplate === template ? ability : template.abilities[0]);
+		if (templateAbility === 'Huge Power' || templateAbility === 'Pure Power') {
+			bst += (evs.atk > evs.spa ? template.baseStats.atk : template.baseStats.spa) + 45;
+		} else if (templateAbility in {'Guts':1, 'Parental Bond':1, 'Hustle':1, 'Mega Launcher':1, 'Protean':1}) {
```

Comment:
Guts and Hustle has negative effects, losing health and inaccuracy respectively, so I wouldn't put that on the same level as those.